Wikipedia/Code/ArticleViewController+References.swift

```diff
@@ -6,7 +6,38 @@ extension ArticleViewController: UIPageViewControllerDelegate {
     }
 }
 
-extension ArticleViewController: WMFReferencePageViewAppearanceDelegate, ReferenceBackLinksViewControllerDelegate, ReferenceShowing {
+
+extension ArticleViewController: ReferenceBackLinksViewControllerDelegate, WMFReferencePageViewAppearanceDelegate, ReferenceShowing {
+    func referenceViewControllerUserDidTapClose(_ vc: ReferenceViewController) {
+        if vc is ReferenceBackLinksViewController {
+            dismissReferenceBackLinksViewController()
+        } else {
+            dismissReferencesPopover()
+        }
+    }
+
+    func referenceBackLinksViewControllerUserDidNavigateTo(referenceBackLink: ReferenceBackLink, referenceBackLinksViewController: ReferenceBackLinksViewController) {
+        scroll(to: referenceBackLink.id, centered: true, highlighted: true, animated: true)
+        dispatchOnMainQueueAfterDelayInSeconds(1.0) {
```

Comment:
Does this delay exist to wait for the scroll to finish? If so, could this block be sent as the completion to `scroll(to:`? Also, to avoid a retain cycle (or calling this after the VC is gone) should the block weakly capture self?

Reply:
Updated, thank you for these. I looked at the function below, hadn't realized `scroll to` could take a completion. And `weak self`s are sometimes my achilles heal. Good catch, thank you!

Reply:
Good call - I clearly replicated the wrong part of the subsequent function. And thanks for catching the missing `weak self` - that's often my achilles heal.